### ISSUE TITLE
"Refinement Filters" field in the Result Web Part don't work as expected

### ISSUE THREAD

**Issue by AlfreMoreno** [microsoft-search/pnp-modern-search#123]
**Version used**
3.10.0

**Describe the bug**
"Refinement Filters" field in the Result Web part does not behave like filters from URL. The documentation does not explain how to use this field, I expect to be similar to URL filters.

- The same JSON structure is not allowed (It fails)
- When using only one filter, the filter is applied but not reflected in the Refiner Web Part.
- When the filter is applied, there is no way to disable it in the filters webpart . The filter is always applied regardless of whether the user modifies the values in the filter webpart.

**Expected behavior**
- Filters in this field behave just like filters in the URL and are applied when page load
- Filters will be reflected in the Refiner Web Part.
- If the user modifies the values in the Refiner Web Part filters will be updated regardless of the value of the default field.

**Screenshots**
Some filters that worked after some testing:
Using Token:
![Filters2](https://user-images.githubusercontent.com/22582515/75802098-2781c300-5d4a-11ea-96f8-215846f2af92.PNG)

Using Value:
![Filters1](https://user-images.githubusercontent.com/22582515/75802085-218be200-5d4a-11ea-8f54-11f6004a13cc.PNG)

Everything else I tried in this field failed.

Unable to clear or Update Filters:
![Unable to Remove Filter](https://user-images.githubusercontent.com/22582515/75802134-336d8500-5d4a-11ea-88b1-27a6a81a3e51.PNG)

All  API call send the filter applied:
![Filter always applied](https://user-images.githubusercontent.com/22582515/75802181-4b450900-5d4a-11ea-9b74-c4a0a756617e.PNG)

**Desktop (please complete the following information):**
 - Browser Chrome]

**Additional context**
Looking into the Code, the field value is passed to the SearchService but don't follow the same logic that the URL Filters 

![image](https://user-images.githubusercontent.com/22582515/75802631-0372b180-5d4b-11ea-83df-f98a0aedbaf9.png)
 
Maybe we can look for filters in the query string, if there is none then we look in the refinementFilters property and apply the same logic.

**Comment by FranckyC** [comment]
Hi @AlfreMoreno , the refinement filters option is not like the URL deep links. If you already used the Search Query Tool, this is basically the same, it allows you to "refine" a query by default if, for whatever reason, you can't do this using the query text and query template (for instance filtering a specific range of time using an FQL expression). In summary, you probably don't need it :). The field supports KQL and FQL expression and the value will be passed to the API as is. It means, you don't need to HEX encode the value.

Also the doc mentions 

` Unlike URL filters params, these won't select the default values in the Search Filters WP if connected.`

**Comment by AlfreMoreno** [comment]
Hi @FranckyC , thanks for your reply. I understand now what is the purpose of that field, I was confused.

Is there any way to configure default filters in the webparts, similar to the URL Filters? I'm trying to configure something like  https://github.com/microsoft-search/pnp-modern-search/pull/110.

For example, I have some Search Pages where I need to search among thousands of items, some of them are active and some of them are inactive. I need to set the search to return active items by default. But the users will be allowed to modify the filters in the Filter webpart. for example, they can change the filters to show only inactive items or all items. 

Basically is similar to the URL filters but defined somehow in the webparts, so when the users land on that page the filters are applied.

Is there any way to do this?

**Comment by FranckyC** [comment]
> when the users land on that page the filters are applied.

Why not using the URL pre-selected filters then because this is exactly the purpose of this feature? Using a specific flag for each refiner will probably conflict with current implementation and honnestly I don't get the use case here.

**Comment by AlfreMoreno** [comment]
Hi @FranckyC , 

After the Url Filters addition, there is no need for individual flags anymore. What I mean is that having a field where we can drop some default filters like `[{"n":"Archive","t":["False"]}]` can be useful. 
We can use exactly the same logic that URL Filters.

>  I have some Search Pages where I need to search among thousands of items, some of them are active and some of them are inactive. I need to set the search to return active items by default. But the users will be allowed to modify the filters in the Filter web part. for example, they can change the filters to show only inactive items or all items.

In this use case, if we add a field to the WP for this purpose, We can set the default filters and the filters will always be applied when the page loads if the query string does not contain filters. 

On the other hand, using the URL filters approach, we will need to set the query string in all links to the search page across the site. Also, the user can clean the query string and navigate directly to the page without filters.

I know this is more an addition, please let me know if this makes sense to you and I can create a PR with the new field. I think it's not a breaking change, please give me your feedback.


Thanks

**Comment by FranckyC** [comment]
Hi @AlfreMoreno I understand the use case. We could of course use a specific field for that in the search results WP. My concern is having too many options to set the default filters in the WP (via URL, via a specific field, using refinement filters) which can be confusing for the user. If we do that, the documentation should be pretty clear about each option.

I will think about this . Maybe by using the same concept as the search box with dynamic data source for filters (either use a static text value or use a query string parameter).

**Comment by AlfreMoreno** [comment]
> I will think about this . Maybe by using the same concept as the search box with dynamic data source for filters (either use a static text value or use a query string parameter).

I like it, Thank you @FranckyC 

Please let me now if you want me to help with some implementation.
